backend/infrahub/events/generator.py

```diff
@@ -68,4 +70,29 @@ async def generate_node_mutation_events(
 
     group_parser = GroupNodeMutationParser(db=db, branch=branch)
     group_events = await group_parser.group_events_from_node_actions(events=events)
-    return events + group_events
+
+    specific_events: list[InfrahubEvent] = []
+    if (kind := node.get_kind()) in [
+        InfrahubKind.CHANGETHREAD,
+        InfrahubKind.OBJECTTHREAD,
+        InfrahubKind.ARTIFACTTHREAD,
+        InfrahubKind.FILETHREAD,
+    ]:
+        proposed_change: CoreProposedChange = await node.change.get_peer(db=db, peer_type=CoreProposedChange)  # type: ignore[attr-defined]
```

Comment:
As a follow up I think for newly created threads we should be able to access the peer proposed change ID from the changelog and avoid this DB lookup. It would be problematic for update events though. I'd suggest we get this merged first and then evaluate this small change.